cou/apps/auxiliary.py

```diff
@@ -204,6 +204,23 @@ def _get_change_require_osd_release_step(self) -> PreUpgradeStep:
 class OvnPrincipal(AuxiliaryApplication):
     """Ovn principal application class."""
 
+    def upgrade_plan_sanity_checks(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> None:
+        """Run sanity checks before generating upgrade plan.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan, defaults to None
+        :type units: Optional[list[Unit]], optional
+        :raises ApplicationError: When enable-version-pinning is True
```

Comment:
not-blocking: This function can raise more than ApplicationError, I know it's coming from inherited class, but still I think it should be mention here. I'm doing the same thing [here](https://github.com/canonical/charmed-openstack-upgrader/pull/331/files#diff-a29275aa9b0cacf5c69098e1afcce54ff8c5c0c6cd0480d12a5b483939047b34R217).

Reply:
I don't see much benefit on doing this because of the maintainability. If we had a tool that help us track this kind of thing I would be favorable of doing it, but right now we make a change it's very easy to have docstrings not updated.